### PR TITLE
Event api

### PR DESCRIPTION
**Pull request by mlynch5187**
## All Submissions:

Resolves #

* [x] Merge the target branch into the PR branch. Fix any conflicts that might appear.

### New Feature Submissions:

* [x] Have you added an explanation of what your changes do?
* [x] Have you lint your code locally prior to submission?

### Testing:

* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?
* [x] All new and existing tests passed.
* [x] Have you run `rubocop` and fixed offenses?

**Test coverage** =  94.65%
--------------------------------------------------------------------------------
### Change Log:

- Events can now be classified by genre
- Events are now parsed successfully
- Update user trip controller create method, start populate method in trip model
- Add webmock stubbing to user trips edit spec
- Create populate method in trip model, self.events_from in event_serializer; events are now chosen for new trips
- All Rubocop offenses resolved

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-# Include: app/controllers/**/*.rb
-Rails/LexicallyScopedActionFilter:
-  Exclude:
-    - 'app/controllers/user/base_controller.rb'
-
-# Offense count: 1
-# Configuration parameters: MinBodyLength.
-Style/GuardClause:
-  Exclude:
-    - 'app/controllers/user/trips_controller.rb'
```
Comment (tylertomlinson) [inline .rubocop_todo.yml:20]: Should be resolved or moved into `.rubocop.yml`

File: spec/features/user/trips/new_spec.rb [hunk 2]
```diff
@@ -5,6 +5,15 @@
     it 'I can see a form to make a new trip' do
       @user = create(:user)
       allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
+      file = File.read('spec/fixtures/response.json')
```
Comment (tylertomlinson) [inline spec/features/user/trips/new_spec.rb:8]: Nice work abstracting from test so use again if neeed

File: spec/features/user/trips/new_spec.rb [hunk 3]
```diff
@@ -26,6 +35,17 @@
...
+      # within('.trip_events") do
+      #   expect(page'.to have_conten'('india'')
+      #   expect(page).to have_content('mexican')
+      #   expect(page).to have_content('italian')
+      #   expect(page).to have_content('music')
+      #   expect(page).to have_content('sports')
+      #   expect(page).not_to have_content('french')
+      #   expect(page).not_to have_content('tacos')
+      #   expect(page).not_to have_content('pool parties')
+      # end
```
Comment (tylertomlinson) [inline spec/features/user/trips/new_spec.rb:48]: Can we remove these or uncomment?

File: .rubocop_todo.yml (hunk 1 above)
Comment (mlynch5187) [reply .rubocop_todo.yml:20]: ok

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -12,7 +12,6 @@
 # the additional setup, and require it from the spec files that actually need
 # it.
 #
-# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
```
Comment (tylertomlinson) [inline spec/spec_helper.rb:15]: I agree this is not needed 🥇

File: spec/features/user/trips/new_spec.rb (hunk 3 above)
Comment (elguapogordo) [inline spec/features/user/trips/new_spec.rb:48]: can be moved to another test or removed later

**Pull request merged**